internal/collector/erc20_transfer_event.go

```diff
@@ -83,19 +107,46 @@ func (col *ERC20TransferEvent) Collect(ch chan<- prometheus.Metric) {
 		eventsLeft := it.Next()
 		if !eventsLeft && it.Error() == nil {
 			// Finished reading events, advance lastQueriedBlock and publish histogram data
-			ch <- prometheus.MustNewConstHistogram(col.desc, count, sum, nil)
-
-			col.lastQueriedBlock = currentBlockNum
+			ch <- prometheus.MustNewConstHistogram(col.desc, count, sum, nil, info.Address, info.Symbol)
 			return
 		} else if !eventsLeft {
-			wErr := errors.Wrap(err, "failed to read transfer event")
+			wErr := errors.Wrapf(err, "failed to read transfer event for contract=[%s]", info.Address)
 			ch <- prometheus.NewInvalidMetric(col.desc, wErr)
 			return
 		}
 		te := it.Event
 
 		value, _ := new(big.Float).SetInt(te.Tokens).Float64()
 		count += 1
-		sum += value
+		sum += value / math.Pow10(int(info.Decimals))
 	}
+
+}
+
+func (col *ERC20TransferEvent) Collect(ch chan<- prometheus.Metric) {
+	col.collectMutex.Lock()
+	defer col.collectMutex.Unlock()
+
+	currentBlockNumber, err := col.bnGetter.BlockNumber(context.Background())
+	if err != nil {
+		wErr := errors.Wrap(err, "failed to get current block number")
+		ch <- prometheus.NewInvalidMetric(col.desc, wErr)
+		return
+	}
+	// INV: currentBlockNum >= lastQueriedblock
+
+	wg := sync.WaitGroup{}
+	for contrInfo, client := range col.contractClients {
```

Comment:
`col.contractClients` has pointers for both the keys and values. Any case in which this could cause panics due to nil pointers?